src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1603,24 +1597,76 @@ protected override void TransformManagedToNative(ILCodeStream codeStream)
             // ANSI marshalling. Allocate a byte array, copy characters
             //
 
-#if READYTORUN
             var stringToAnsi =
-                Context.SystemModule.GetKnownType("System.StubHelpers", "AnsiBSTRMarshaler")
+                Context.SystemModule.GetKnownType("System.StubHelpers", "CSTRMarshaler")
                 .GetKnownMethod("ConvertToNative", null);
+
+            bool bPassByValueInOnly = In && !Out && !IsManagedByRef;
+            ILLocalVariable localBuffer = emitter.NewLocal(Context.GetWellKnownType(WellKnownType.IntPtr));
+            if (bPassByValueInOnly)
```

Comment:
is there a reason why this logic cant move to `System.StubHelpers.CSTRMarshaler` ?

Reply:
Yes, it would be nice to have majority of the logic in C#, and minimize amount of the hand-emitted IL.